tensorflow_similarity/augmenters/barlow.py

```diff
@@ -30,9 +31,8 @@ def augment_barlow(
     blur_min_sigma=0,
     blur_max_sigma=1,
     solarize_probability=0.2,
-    solarize_thresh=10
+    solarize_thresh=10,
```

Comment:
This is the only problematic parameter, as the value depends on the range of the image.  To make solarize work, I think we need the user to pass us the expected min and max image values and a threshold.  Requiring the user to pass this avoids the need for us to guess at what the correct range should be.

Something like the following should work for solarized for all value ranges ([0,1], [0, 255], or even [-127.5, 127.5]):

def solarize(img, min, max, thresh):
  midpoint = (max - min)/2.
  adjusted_img = img - min - midpoint
  adjusted_thresh = thresh - min - midpoint
  img = np.where(adjusted_img < adjusted_thresh, adjusted_img, -adjusted_img)
  img = img + min + midpoint
  return img

wdyt?

Reply:
sounds good. Will try to make these changes tomorrow

Reply:
Why don't we do something like:

```py
def solarize(img, min, max, thresh):
     img = np.where(img < thresh, img, max - img)
```

instead? why do we have to adjust the images?

Reply:
The simpler version is cleaner but I think it only supports positive values. If a client passes in an image with min = -127.5, max = 127.5 and thresh = -117.5, then we'll end up mapping -117.5 to (125.5 - -117.5) = 243. 

I suppose it doesn't matter though if the other transforms only support positive values as well.

Reply:
Okay, good to know. BTW, have noticed that the tests failed, but not quite sure why.

Reply:
I've come across this before. Looks like the TF version updated and there were some changes to the tf.keras.application models. Seems like they added an extra layer. We can just update the expected number in the test, and I should make those tests less brittle at some point.

Reply:
Let's just merge it and I can fix them up on my end. 👍

Reply:
Okay sure, you can merge this PR then :)